src/Core/src/Core/Extensions/VisualTreeElementExtensions.cs

```diff
@@ -145,11 +145,14 @@ static List<IVisualTreeElement> GetVisualTreeElementsWindowsInternal(IVisualTree
 
 			if (uiElement != null)
 			{
-				var uniqueElements = findChildren(uiElement).Distinct();
-				var viewTree = visualElement.GetVisualTreeDescendants().Where(n => n is IView view && view.Handler is not null).Select(n => new Tuple<IView, object?>((IView)n, ((IView)n).ToPlatform()));
-				var testList = viewTree.Where(n => uniqueElements.Contains(n.Item2)).Select(n => n.Item1);
-				if (testList != null && testList.Any())
-					visualElements.AddRange(testList.Select(n => (IVisualTreeElement)n));
+				var uniqueElements = findChildren(uiElement).ToHashSet();
+
+				var testList = visualElement.GetVisualTreeDescendants()
+					.Where(n => n is IView view && view.Handler is not null && uniqueElements.Contains(view.ToPlatform()))
+					.ToList();
+
+				if (testList.Count > 0)
+					visualElements.AddRange(testList);
```

Comment:
What if you removed all the System.Linq usage here? Is it possible to get the same result using a plain `foreach` loop?